toolchain/lower/testdata/builtins/method_vs_nonmethod.carbon

```diff
@@ -0,0 +1,24 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+
+fn AddNonmethod(a: i32, b: i32) -> i32 = "int.add";
```

Comment:
Just to note though, but not having a `__builtin` or anything like that here makes this feel like it's really something intended for developers at the large, even if poorly documented.

Reply:
Perhaps we can restrict this syntax to the `Core` package. (I think I'd also be fine with starting all the builtin names with `builtin.` or `core.` or `lang.` or something.)

We could consider having a mode for the toolchain that says "I'm compiling the core package itself" that turns off the automatic dependency on `Core`, enables defining builtins, and causes lookup for `Core` names to look in the current package scope instead? (But not in this PR.)

Reply:
Yeah, that might be worthwhile as enforcement. Although, from your suggestions -- maybe "Internal."?